From 0302fcd912efc9b45eb1a5f2af4dc5f94f6b6c28 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 24 Feb 2026 15:06:55 +0100
Subject: [PATCH] Revert "udata: Store u32 udata values in Big Endian"

JIRA: https://issues.redhat.com/browse/RHEL-119650
Upstream Status: libnftnl commit b587e4a7bbc8d6b574545617f2639dbd8905ec5d

commit b587e4a7bbc8d6b574545617f2639dbd8905ec5d
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Jan 29 14:24:49 2026 +0100

    Revert "udata: Store u32 udata values in Big Endian"

    This reverts commit f20dfa7824860a9ac14425a3f7ca970a6c981597.

    This change to payload (interpretation) is problematic with package
    updates at run-time: The new version might trip over userdata in the
    running ruleset, avoid this.

    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/udata.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/udata.c b/src/udata.c
index 8cf4e7c..a195657 100644
--- a/src/udata.c
+++ b/src/udata.c
@@ -8,7 +8,6 @@
 #include <udata.h>
 #include <utils.h>
 
-#include <arpa/inet.h>
 #include <stdlib.h>
 #include <stdint.h>
 #include <string.h>
@@ -101,9 +100,7 @@ EXPORT_SYMBOL(nftnl_udata_put_u32);
 bool nftnl_udata_put_u32(struct nftnl_udata_buf *buf, uint8_t type,
 			 uint32_t data)
 {
-	uint32_t data_be = htonl(data);
-
-	return nftnl_udata_put(buf, type, sizeof(data_be), &data_be);
+	return nftnl_udata_put(buf, type, sizeof(data), &data);
 }
 
 EXPORT_SYMBOL(nftnl_udata_type);
@@ -131,7 +128,7 @@ uint32_t nftnl_udata_get_u32(const struct nftnl_udata *attr)
 
 	memcpy(&data, attr->value, sizeof(data));
 
-	return ntohl(data);
+	return data;
 }
 
 EXPORT_SYMBOL(nftnl_udata_next);
